File: EventNotifier/Data/Entities/EventEntity/Event.cs
using EventNotifier.Data.Entities.CategoryEntity;
using EventNotifier.Data.Entities.FormatEntity;

namespace EventNotifier.Data.Entities
{
    public class Event
    {
        public int? Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int Year { get; set; }
        public Format? Format { get; set; }
        public int? FormatId { get; set; }
        public Category? Category { get; set; }
        public int? CategoryId { get; set; }
        public ICollection<SelectedEvents>? SelectedEvents { get; set; }
    }
}

File: EventNotifier.Client/src/app/demo/models/event.ts
import { Category } from "./category";
import { Format } from "./format";

export interface Event {
    id?: number;
    name?: string;
    description?: string;
    startDate?: Date;
    endDate?: Date;
    year?: number;
    format?: Format;
    category?: Category;
    photo?: Uint8Array[];
}
